packages/viewer/src/components/viewer-pin-group/vertex-pin-group.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  Component,
+  Fragment,
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+import { Dimensions, Matrix4, Point, Vector3 } from '@vertexvis/geometry';
+import classNames from 'classnames';
+
+import { Viewport } from '../..';
+import { PinController } from '../../lib/pins/controller';
+import { isDefaultPin, isTextPin, Pin } from '../../lib/pins/entities';
+import { PinModel } from '../../lib/pins/model';
+import { translatePointToScreen } from '../viewer-markup/utils';
+import { getClosestCenterToPoint } from './utils';
+
+interface ComputedPoints {
+  pinPoint: Point.Point;
+  labelPoint?: Point.Point;
+}
+@Component({
+  tag: 'vertex-viewer-pin-group',
+  styleUrl: 'vertex-pin-group.css',
+  shadow: false,
+})
+export class ViewerPinGroup {
+  /**
+   * The pin to draw for the group
+   */
+  @Prop({ mutable: true })
+  public pin?: Pin;
+
+  /**
+   * The local matrix of this element.
+   */
+  @Prop({ mutable: true, attribute: null })
+  public matrix: Matrix4.Matrix4 = Matrix4.makeIdentity();
+
+  /**
+   * Projection view matrix used for computing the position of the pin line
+   */
+  @Prop()
+  public projectionViewMatrix: Matrix4.Matrix4 = Matrix4.makeIdentity();
+
+  /**
+   * The dimensions of the canvas for the pins
+   */
+  @Prop({ mutable: true })
+  public elementBounds?: DOMRect;
+
+  /**
+   * The model that contains the entities and outcomes from performing pin operations
+   */
+  @Prop()
+  public pinModel: PinModel = new PinModel();
+
+  /**
+   * The controller that drives behavior for pin operations
+   */
+  @Prop()
+  public pinController?: PinController;
```

Comment:
Might be nice to auto-create this controller if none is specified similar to the `vertex-viewer-measurement-precise` element. Otherwise it might not be very clear how to go about creating one